src/btree/bt_curprev.c

```diff
@@ -792,16 +795,17 @@ __wt_btcur_prev(WT_CURSOR_BTREE *cbt, bool truncating)
             LF_SET(WT_READ_VISIBLE_ALL);
 
         /*
-         * If we are running with snapshot isolation, and not interested in returning tombstones, we
-         * could potentially skip pages. The skip function looks at the aggregated timestamp
-         * information to determine if something is visible on the page. If nothing is, the page is
-         * skipped.
+         * If we are running with snapshot isolation, have a snapshot, and are not interested in
+         * returning tombstones, we could potentially skip pages. The skip function looks at the
+         * aggregated timestamp information to determine if something is visible on the page. If
+         * nothing is, the page is skipped.
          */
         if (!F_ISSET(&cbt->iface, WT_CURSTD_KEY_ONLY) &&
           session->txn->isolation == WT_ISO_SNAPSHOT &&
+          F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT) &&
```

Comment:
Looks like this was missed on [7.0](https://github.com/mongodb/mongo/commit/3f041b19cc07d6225b8928b2318a5334caa74175#diff-22e87c7ad25711adac39c7b010e2b93c8d7b0a38d82271fbda34199743766872R887)

Reply:
It is the only change required in 6.0 due to the checkpoint cursors that are not supported, but there exists so many tests that test checkpoint cursors.